x/harvest/keeper/borrow.go

```diff
@@ -91,5 +88,18 @@ func (k Keeper) calculateUSDValue(ctx sdk.Context, amount sdk.Int, denom string)
 	if err != nil {
 		return sdk.ZeroDec(), sdkerrors.Wrapf(types.ErrPriceNotFound, "no price found for market %s", moneyMarket.SpotMarketID)
 	}
-	return sdk.NewDecFromInt(amount).Quo(sdk.NewDecFromInt(moneyMarket.ConversionFactor).Mul(assetPriceInfo.Price)), nil
+	return sdk.NewDecFromInt(amount).Quo(sdk.NewDecFromInt(moneyMarket.ConversionFactor)).Mul(assetPriceInfo.Price), nil
+}
+
+func (k Keeper) getBorrowableAmountForDeposit(ctx sdk.Context, deposit types.Deposit) (sdk.Dec, error) {
+	moneyMarket, found := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
```

Comment:
One thing to keep track of is how many calls we are making to GetMoneyMarket per borrow. Don't see a need to refactor in this PR,  but that's probably the most expensive read operation we make, so minimizing those calls is a valid goal.